.gitlab-ci.yml

```diff
@@ -218,6 +207,90 @@ build_windows_msi_x64:
     paths:
     - .omnibus/pkg
 
+
+
+#
+# image_build
+#
+
+
+# build the agent6 image
+build_agent6:
+  stage: image_build
+  image: 486234852809.dkr.ecr.us-east-1.amazonaws.com/devtools/dind/build:latest
+  tags: [ "runner:dind", "size:large" ]
+  script: [ "# noop" ]
+  variables:
+    DD_DIND_BUILD_CONTEXT: Dockerfiles/agent6
+    DD_DIND_IMAGE: 486234852809.dkr.ecr.us-east-1.amazonaws.com/datadog-agent/agent:$CI_COMMIT_SHA
+    DD_DIND_ARTEFACTS: "true"
+
+# build the dogstatsd image
+build_dogstatsd:
+  stage: image_build
+  dependencies:
+    - build_dogstatsd_static-deb_x64
+  image: 486234852809.dkr.ecr.us-east-1.amazonaws.com/devtools/dind/build:latest
+  tags: [ "runner:dind", "size:large" ]
+  script: [ "# noop" ]
+  variables:
+    DD_DIND_BUILD_CONTEXT: Dockerfiles/dogstatsd/alpine
+    DD_DIND_IMAGE: 486234852809.dkr.ecr.us-east-1.amazonaws.com/datadog-agent/dogstatsd:$CI_COMMIT_SHA
+    DD_DIND_ARTEFACTS: "true"
+
+#
+# image_deploy
+#
+agent6_docker_hub:
```

Comment:
Lots of duplication here, I think you should use [YAML anchors](https://learnxinyminutes.com/docs/yaml/) to create a `.docker_hub_template` step and inherit from it, to reduce duplication. https://docs.gitlab.com/ee/ci/yaml/#anchors

Reply:
Nice, didn't know about this, done